x-pack/plugins/fleet/public/applications/integrations/sections/epm/screens/detail/settings/install_button.tsx

```diff
@@ -35,16 +37,21 @@ export function InstallButton(props: InstallationButtonProps) {
   const getPackageInstallStatus = useGetPackageInstallStatus();
   const { status: installationStatus } = getPackageInstallStatus(name);
 
-  const isInstalling = installationStatus === InstallStatus.installing;
+  const [isFleetSetupInProgress, setFleetSetupInProgress] = useState<boolean>(false);
+
+  const isInstalling = installationStatus === InstallStatus.installing || isFleetSetupInProgress;
   const [isInstallModalVisible, setIsInstallModalVisible] = useState<boolean>(false);
 
   const toggleInstallModal = useCallback(() => {
     setIsInstallModalVisible(!isInstallModalVisible);
   }, [isInstallModalVisible]);
 
-  const handleClickInstall = useCallback(() => {
-    installPackage({ name, version, title });
+  const handleClickInstall = useCallback(async () => {
+    setFleetSetupInProgress(true);
     toggleInstallModal();
+    await sendPostFleetSetup({ forceRecreate: false });
```

Comment:
We may want to handle the error happening in `sendPostFleetSetup` we probably want to close the install modal and  show the error in a toast notification as we do in other place what do you think?

Reply:
good point! do you think the installPackage should continue if setup fails? to fall back to the existing logic

Reply:
Sure we can just catch the error add a toast and try to install the package 👍

Reply:
added error toast, though it shows up for a short time only by default

simulated error by throwing error on backend:
![image](https://user-images.githubusercontent.com/90178898/146398552-c9d27f30-3849-46ff-a3a7-a683d5457b7c.png)

![image](https://user-images.githubusercontent.com/90178898/146398670-d2ee045f-f610-4d45-8f79-76c52beda053.png)